src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbBatchCommand.cs

```diff
@@ -1068,6 +1068,17 @@ private async ValueTask UpdateParameterValuesAsync(Descriptor descriptor, int ba
 				{
 					batchParameter.NullFlag = 0;
 
+					if (commandParameter.Charset > 0 && Charset.TryGetByType(commandParameter.Charset, out var charset))
```

Comment:
There should probably be a plan for case when the charset is not found. Maybe this is point for having `Charset` and `FbCharset` closer together.